### PR TITLE
Add Amazon Q Developer docs

### PR DESCRIPTION
**Pull request by arundna**
#### Description of changes:
Add Amazon Q Developer docs for Gen 2

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  return {
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+[Amazon Q Developer](https://aws.amazon.com/q/) re-imagines the experience across the entire Software Development LifeCycle (SDLC), making it faster and easier for developers and IT professionals to explore and understand problems and codebase, plan and develop a solution, build and deploy their apps, or modernize legacy workloads. Q Developer in the IDE provides [Inline completions (former Amazon CodeWhisperer)](https://docs.aws.amazon.com/codewhisperer/latest/userguide/what-is-cwspr.html) - an AI-powered productivity tool that generates code suggestions ranging from snippets to full functions in 
+real time in the IDE based on your comments and existing code. When you start typing out singles of code or comments, Inline completions
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:31]:
```suggestion
[Amazon Q Developer](https://aws.amazon.com/q/) re-imagines the experience across the entire Software Development LifeCycle (SDLC), making it faster and easier for developers and IT professionals to explore and understand problems and codebase, plan and develop a solution, build and deploy their apps, or modernize legacy workloads. Q Developer in the IDE provides [Inline completions (former Amazon CodeWhisperer)](https://docs.aws.amazon.com/codewhisperer/latest/userguide/what-is-cwspr.html) - an AI-powered productivity tool that generates code suggestions ranging from snippets to full functions in real time in the IDE based on your comments and existing code. When you start typing out singles of code or comments, Inline completions
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+For information about authenticating with Amazon Q Developer - Inline completions, see [Authenticating with Amazon Q Developer - Inline completions and AWS Toolkit](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth).
+
+
+### Setting up your authentication method
+
+- [Builder ID](https://docs.aws.amazon.com/signin/latest/userguide/sign-in-aws_builder_id.html) (used with AWS Toolkit and VS Code or JetBrains) requires only an email address. To use it, you don't even need an AWS account.
+
+- [IAM Identity Center](https://docs.aws.amazon.com/singlesignon/latest/userguide/what-is.html) requires setup by your enterprise administrator.
+
+- [IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) credentials are used within your AWS account to regulate access to, and between, various AWS services.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:79]:
```suggestion
- [Builder ID](https://docs.aws.amazon.com/signin/latest/userguide/sign-in-aws_builder_id.html) (used with AWS Toolkit and VS Code or JetBrains) requires only an email address. To use it, you don't even need an AWS account.
- [IAM Identity Center](https://docs.aws.amazon.com/singlesignon/latest/userguide/what-is.html) requires setup by your enterprise administrator.
- [IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) credentials are used within your AWS account to regulate access to, and between, various AWS services.
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.
+
+![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/VSCode.png)
+
+</Block>
+</BlockSwitcher>
+
+## Setup a AWS Amplify Gen 2 project
+
+You can quickly setup an Amplify Gen 2 project using one of our [Quickstart](/start) guides. In this example, we will setup a [Next.js App Router (Client Components)](https://next-release-main.d1ywzrxfkb9wgg.amplifyapp.com/javascript/start/quickstart/nextjs-app-router-client-components/).
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:108]: nit for the preview link

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+`create-amplify` provides the scaffolding of a `amplify/data/resource.ts` file, which is ready to deploy.
+
+## Use Q Developer - Inline completions in your Gen 2 project
+
+Amplify generates two folders in our backend directory, `auth` and `data` which contain TypeScript CDK definitions for each of these resources.  We’ll build out the schema for our API through the help of Amazon Q Developer - Inline completions capabilities.
+
+**Step 1:** Open `amplify/data/resource.ts` and comment out the default schema for Todo provided.
+
+```ts title="amplify/data/resource.ts"
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:119]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+//     .authorization([a.allow.public()]),
+// });
+
+// ...
+```
+
+**Step 2:** In a new line below the commented schema, enter a comment to generate the schema using natural language. For example, `generate a restaurant model with the following fields: id, name, description, address, image, rating, style. Rating can be a float value. Authorization should allow public.`  Press `Enter` for a new line and wait for Amazon Q Developer to generate Inline completion for your schema.
+
+```ts title="amplify/data/resource.ts"
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:138]:
```suggestion
```

**Review by josefaidt — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Use Amplify with Amazon Q Developer',
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:5]:
```suggestion
  title: 'Use Amazon Q Developer with Amplify',
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+**Note:** You can also trigger Inline completion feature by invoking Amazon Q Developer manually using Option+C keyboard shortcut in VS Code. For more commands, please refer to the Commands tab in the AWS Toolkit extension. 
+
+</Callout>
+
+**Step 4:** Make any required changes to the schema and save the `amplify/data/resource.ts` file. This will trigger a sandbox deployment and your new data model will be deployed 
+
+## Conclusion
+
+Success! You've learned how to use Amazon Q Developer - Inline completions feature with Amplify.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:170]: cut 
```suggestion
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/VSCode.png)
+
+</Block>
+</BlockSwitcher>
+
+## Setup a AWS Amplify Gen 2 project
+
+You can quickly setup an Amplify Gen 2 project using one of our [Quickstart](/start) guides. In this example, we will setup a [Next.js App Router (Client Components)](/[platform]/start/quickstart/nextjs-app-router-client-components/).
+
+`create-amplify` provides the scaffolding of a `amplify/data/resource.ts` file, which is ready to deploy.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:107]: cut
```suggestion
```

**Review by renebrandel — DISMISSED** [review]

**Review by josefaidt — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+![A screenshot of AWS Toolkit for VS Code that displays options with sign in mechanisms for Amazon Q + CodeWhisperer, AWS Explorer, and Amazon CodeCatalyst](/images/gen2/q-developer/VSCode.png)
+
+</Block>
+<Block name="JetBrains">
+
+1. From the AWS Toolkit for JetBrains, select the **Amazon Q + CodeWhisperer** tab.
+1. Under CodeWhisperer, choose **Sign in to get started**. The **AWS Toolkit: Setup Authentication** modal will open.
+1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.
+
+![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/VSCode.png)
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:98]: This screenshot doesn't show Jetbrains, this is showing VSCode.

Comment (reesscot) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:98]:
```suggestion
![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/JetBrains.png)
```

Comment (arundna) [reply src/pages/[platform]/build-a-backend/q-developer/index.mdx]: Thanks for catching this, your suggested changes will fix the img for JetBrains.

**Review by renebrandel — DISMISSED** [review]

**Review by josefaidt — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+## Getting started with Q Developer - Inline completions in VS Code and JetBrains
+
+<BlockSwitcher>
+<Block name="VS Code">
+
+1. From the AWS Toolkit for VS Code, in the AWS pane, under **CodeWhisperer**, choose **Sign in to get started**. The **AWS Toolkit: Add Connection to AWS** tab will open.
+1. Select the **Amazon Q + CodeWhisperer** authentication panel.
+1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.
+
+![A screenshot of AWS Toolkit for VS Code that displays options with sign in mechanisms for Amazon Q + CodeWhisperer, AWS Explorer, and Amazon CodeCatalyst](/images/gen2/q-developer/VSCode.png)
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:89]: nit: I think this is slightly more descriptive:

```suggestion
![A screenshot of AWS Toolkit for VS Code that displays auth config options for Amazon Q + CodeWhisperer, AWS Explorer, and Amazon CodeCatalyst](/images/gen2/q-developer/VSCode.png)
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+
+</Block>
+<Block name="JetBrains">
+
+1. From the AWS Toolkit for JetBrains, select the **Amazon Q + CodeWhisperer** tab.
+1. Under CodeWhisperer, choose **Sign in to get started**. The **AWS Toolkit: Setup Authentication** modal will open.
+1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.
+
+![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/JetBrains.png)
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:98]: nit: I think this is slightly more descriptive:

```suggestion
![A screenshot of AWS Toolkit for JetBrains that displays auth config options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/JetBrains.png)
```

**Review by josefaidt — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**